### PR TITLE
Added support for Dropwizard configuration on dashboard

### PR DESCRIPTION
**Pull request by mbarrientos**
Added support for Dropwizard configuration on dashboard:
- config.yml is now bootstrapped with environment variables.
- REST resources access the endpoint through factories providing config information.

### CODE REVIEW

**Comment by mbarrientos** [comment]
This fixes #48.

File: dashboard/src/main/java/eu/seaclouds/platform/dashboard/DashboardApplication.java [hunk 1]
```diff
@@ -34,16 +36,25 @@ public static void main(String[] args) throws Exception {
 
     @Override
     public void initialize(Bootstrap<DashboardConfiguration> bootstrap) {
+        // Setting configuration from env variables
+        bootstrap.setConfigurationSourceProvider(
+                new SubstitutingSourceProvider(bootstrap.getConfigurationSourceProvider(),
+                        new EnvironmentVariableSubstitutor(false)
```
Comment (andreaturli) [inline dashboard/src/main/java/eu/seaclouds/platform/dashboard/DashboardApplication.java]: why `strict=false` ?

File: dashboard/config.yml [hunk 2]
```diff
@@ -4,4 +4,15 @@ server:
     port: 8000
   adminConnectors:
   - type: http
-    port: 8001
\ No newline at end of file
+    port: 8001
+deployer:
+  host: ${DEPLOYER_HOST}
```
Comment (andreaturli) [inline dashboard/config.yml]: In this way we are still substituting those values using ENV VAR, which is fine but not ideal.
When I suggested to use Dropwizard configuration I was expecting to have a `config.yml.template` where values were replaced by Brooklyn when known, but we'd need a dashboard entity for that, so let's keep it as is.

Probably the idiomatic way in Dropwizard is to define your own `ConfigurationSourceProvider` but not sure.

Comment (mbarrientos) [inline dashboard/config.yml]: The only problem to using Brooklyn yaml templating is that it makes the Dashboard a bit more Brooklyn dependent (which is not a big problem!). Maybe we could support two templates or something similar.

However, this template is OK for building with any kind of distribution ATM. I'll add this feature when the brooklyn dashboard entity is ready, which will be soon.

File: dashboard/src/main/java/eu/seaclouds/platform/dashboard/DashboardApplication.java (hunk 1 above)
Comment (mbarrientos) [inline dashboard/src/main/java/eu/seaclouds/platform/dashboard/DashboardApplication.java]: I let it false to support empty fields on the user/pass parameters.

**Comment by mbarrientos** [comment]
@andreaturli I opened https://github.com/SeaCloudsEU/SeaCloudsPlatform/pull/62 to add brooklyn config support to the actual seaclouds distribution (dashboard only). 
Right now, the dashboard is able to build from env vars when launching as a standalone and with brooklyn config keys when using Brooklyn Entity.

If you consider it ok we can merge this and move forward with https://github.com/SeaCloudsEU/SeaCloudsPlatform/pull/62

**Comment by andreaturli** [comment]
@mbarrientos I think it is really good, let's merge that.

I've a quick look at #62, not sure `usage/installer` is the right place for the new entities (also modaclouds entities should be moved, my bad) but the implementation looks exactly what we need

**Pull request merged**